src/commands.cpp

```diff
@@ -0,0 +1,152 @@
+#include "common_package/commands.hpp"
+
+using namespace common_lib;
+
+/**
+ * @brief Parses and checks a string formatted JSON against a given definition
+ *
+ * - Checks that no undefined keys are in the JSON
+ * - Checks that all required keys exist
+ * - Checks that all values have the correct type
+ *
+ * @note The function can only be used on shallow JSONs. Arrays or encapsulations are not supported and will result in an exception!
+ *
+ * @param json_str String formatted json that should be parsed and checked
+ * @param definition Definition of what keys shall be included and what type they can have
+ * @throws std::runtime_error with an error message why the check failed
+ */
+nlohmann::json CommandDefinitions::parse_check_json(const std::string &json_str, const std::map<std::string, JsonKeyDefinition> definition)
```

Comment:
Referenz
```suggestion
nlohmann::json CommandDefinitions::parse_check_json(const std::string &json_str, const std::map<std::string, JsonKeyDefinition> &definition)
```

Und vllt const:
```suggestion
nlohmann::json CommandDefinitions::parse_check_json(const std::string &json_str, const std::map<std::string, JsonKeyDefinition> definition) const
```

Reply:
Da die Funktion `static` ist funktioniert kein `const` (und macht auch keinen Sinn).

Reply:
Referent ist aber gemacht.